crates/ide_assists/src/handlers/generate_function.rs

```diff
@@ -85,31 +85,18 @@ fn gen_fn(acc: &mut Assists, ctx: &AssistContext) -> Option<()> {
         None => None,
     };
 
-    let function_builder = FunctionBuilder::from_call(ctx, &call, &path, target_module)?;
+    let (target, file, insert_offset) = get_fn_target(ctx, &target_module, call.clone())?;
+    let function_builder = FunctionBuilder::from_call(ctx, &call, &path, target_module, target)?;
     let target = call.syntax().text_range();
-
-    acc.add(
-        AssistId("generate_function", AssistKind::Generate),
-        format!("Generate `{}` function", function_builder.fn_name),
```

Comment:
We seem to be losing that label here, we do want to show `Generate \`{}\` function` when generating functions and method for methods.

Reply:
https://github.com/rust-analyzer/rust-analyzer/pull/9804#discussion_r684778026

I think it was concluded that they should have the same id.

Reply:
This is about the *label*, not the id. The label is being shown to the user and it should be clear whether this creates a function or method. The id should be shared that is correct.

Reply:
> I think generate_function and generate_method should be one thing from the perspective of the user

I think matklad insisted that we should not mention any difference between the two. But if you really want this, I'll change.